### DISCUSSION TITLE
Filter for connectors

### DISCUSSION THREAD

**Discussion by dpolivaev** [freeplane/freeplane#1801] [Announcements]
I implemented a new filter mode which when active is applied only to the connectors so that only such connectors are shown where both source and target nodes satisfy the filter condition.

My specific use case is related to the code explorer mode where I want to unroll connectors representing dependencies from or to some class step by step, and I think other use cases might also arise.

I would appreciate any feedback on the idea.

**Comment by dpolivaev** [comment]
Preview 1.12.1_03

<img width="800" alt="filterToolbar" src="https://github.com/freeplane/freeplane/assets/1558052/68dfe483-1a9f-403d-abf8-cfee1096c8d8">

**Comment by euu2021** [comment]
I created an example, to understand the feature:

Starting with this situation:
![image](https://github.com/freeplane/freeplane/assets/77707706/31c3e2c0-4407-4284-949b-befdd34b310c)

I composed the filter:
![image](https://github.com/freeplane/freeplane/assets/77707706/a3540484-f271-4fbc-b21b-42c7ed224dc7)

There are two nodes that match the filter: ccc and fff.

Now, with the filter connector activated, what happens when the filter is applied is that only the connectors that have both source and target node matching the filter are shown. In the example, it's only this connector, because it connects ccc and fff, which are nodes that match the filter criteria:

![image](https://github.com/freeplane/freeplane/assets/77707706/33609044-b436-4f80-af97-02eb942304d5)

So, notice that only connectors are filtered, and not nodes.

**Reply by dpolivaev** [reply]
Yes, it is exactly what I need.

**Reply by euu2021** [reply]
> Yes, it is exactly what I need.

I know. The example, is just to explain the feature. At first, I did not understand what it does.

**Reply by dpolivaev** [reply]
Yes, I do not know how to explain such things. It is why I write no documentation and Freeplane and its features remain mostly undiscovered.

**Comment by macmarrum** [comment]
When I think about filtering related to Connectors, I come up with 3 main cases:
1. Node Filter
2. Connector Filter
3. Node and Connector Filter

The following examples present the cases:

![2024-04-27_18,57,18](https://github.com/freeplane/freeplane/assets/7197131/e3104db3-c556-43cf-9fc1-cc961915ea3b)

![2024-04-27_18,58,51](https://github.com/freeplane/freeplane/assets/7197131/aa0add2f-3cff-4653-9dc9-9dc8cab358cb)

![2024-04-27_19,00,49](https://github.com/freeplane/freeplane/assets/7197131/c1d86616-f392-4f4d-a5ac-d466796b7107)

![2024-04-27_19,02,24](https://github.com/freeplane/freeplane/assets/7197131/6ffd3047-2ba3-4b81-81f5-01dd87e8180d)

Freeplane 1.11.x supports case 1 – Node Filter (based on Connector properties).
Freeplane 1.12.1-alpha-3 additionally supports case 2 – Connector Filter (based on Node or Connector properties).

I think it would be great for Freeplane to also support case 3.

**Reply by dpolivaev** [reply]
Implemented in preview 1.12.1_04

<img width="814" alt="filterModes" src="https://github.com/freeplane/freeplane/assets/1558052/d7e4ae32-4084-4d27-80bb-c56c6f3d0b6c">
